stan/math/prim/prob/neg_binomial_2_lcdf.hpp

```diff
@@ -29,23 +30,25 @@ return_type_t<T_location, T_precision> neg_binomial_2_lcdf(
   scalar_seq_view<T_n> n_vec(n);
   scalar_seq_view<T_location> mu_vec(mu);
   scalar_seq_view<T_precision> phi_vec(phi);
-
+  size_t size_n = size(n);
   size_t size_phi_mu = max_size(mu, phi);
+
+  for (size_t i = 0; i < size_n; i++) {
+    if (n_vec[i] < 0) {
+      return LOG_ZERO;
+    }
+  }
+
   VectorBuilder<true, return_type_t<T_location, T_precision>, T_location,
                 T_precision>
       phi_mu(size_phi_mu);
   for (size_t i = 0; i < size_phi_mu; i++) {
     phi_mu[i] = phi_vec[i] / (phi_vec[i] + mu_vec[i]);
   }
 
-  size_t size_n = size(n);
   VectorBuilder<true, return_type_t<T_n>, T_n> np1(size_n);
   for (size_t i = 0; i < size_n; i++) {
-    if (n_vec[i] < 0) {
-      return log(0.0);
-    } else {
-      np1[i] = n_vec[i] + 1.0;
-    }
+    np1[i] = n_vec[i] + 1.0;
```

Comment:
[optional]
These don't need the `.0` --- just `1` is fine.  The  compiler will promote.